web_hierarchy/static/src/js/web_hierarchy_view.js

```diff
@@ -0,0 +1,98 @@
+ /*
+ * Copyright 2019 Kevin Kamau <kevkamaa96@gmail.com>
+ * License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+ */
+
+ /*eslint no-undef: "google", "drawChart" */
+
+odoo.define('web.HierarchyView', function (require) {
+    'use strict';
+
+    var core = require("web.core");
+    var View = require('web.View');
+    var _lt = core._lt;
+    var qweb = core.qweb;
+
+    var HierarchyView = View.extend({
+        template: 'HierarchyView',
+        icon: "fa-sitemap",
+        searchable: false,
+        mobile_friendly: true,
+        require_fields: true,
+        display_name: _lt('Hierarchy'),
+        render_chart: function (size) {
+            var self = this;
+            var size_val = !size ? "medium": size; /*eslint no-negated-condition: "error"*/
+            console.log(size_val);
+            var parent_name = this.options.action.display_name;
+            google.charts.load('current', {packages: ["orgchart"]});
+            google.charts.setOnLoadCallback(drawChart);
+
+            function drawChart() {
+                var vals = [];
+                var data = new google.visualization.DataTable();
+                data.addColumn('string', 'ID');
+                data.addColumn('string', 'Name');
+                data.addColumn('string', 'ToolTip');
+                self.dataset.read_slice(self.fields_view.fields, {}).then(
+                    function (results) {
+                        var root = parent_name || "Root";
+                        vals.push([{v: root, f: root}, '', 'root element']);
+                        for (var i = 0; i < results.length; i++) {
```

Comment:
Here can use cached loop
```suggestion
                        var l = results.length;
                        for (var i = 0; i < l; ++i) {
```

Reply:
done